### PR TITLE
Adjusted return types of API aliases, removed ambiguous methods

### LINKED ISSUE #1651: Remove some methods from API
We should be careful when having multiple methods with same name but different argument type. This was also discussed in #721.

Therefore I suggest to keep only the of(T), of(T...)-like methods and remove the ofAll()-like methods.

(Maps will have of(KVKVKV), sorted collections will have a variant with a Comparator argument)

### LINKED ISSUE #1662: Changes to API aliases
**Conventions:**

* Match/Case are written upper-case because `case` is reserved
* For is written upper case because `for` is reserved
* Type aliases are written upper-case because they represent constructors
* n-ary type aliases are written upper-case, e.g. Tuple(...) for Tuple1, ..., Tuple8
* helper functions are written lower-case (println, unchecked, ...). Exception: TODO

**Goal of this issue:**

* To solve ambiguities we need to remove all aliases for collections but only keep these:
    * Name()
    * Name(T)
    * Name(T...)
* Maps are a special case, here we have pairs (see also #1608):
    * Name()
    * Name(K1, V1), ... Name(K1, V1, ..., K10, V10)
* The sorted collections also have variants with a preceeding Comparator argument in addition to the two points above.
* The methods Unchecked(CheckedFunction) need to be written lower-case because they not represent a type. These methods are helper functions.
* The aliases for Future differ from other aliases and are also ambiguous. E.g. we currently have Future(T) and Future(Throwable) but Option(T), Some(T) and None() / Try(CheckedSupplier), Success(T) and Failure(Throwable). We need s.th. like Future(CheckedSupplier), Successful(T), Failed(Throwable) - accordingly to the existing factory methods. **Update:** In Scala there is only Future(=&gt; computation) and no types Successful and Failed. So we will keep Future(Supplier) and Future(T) and throw away the ambiguous Future(Throwable).
* more?

**Changes that will be done in other issues:**

* Pattern0, ..., Pattern8 will be hidden, we currently only propagate Pattern to the outside (see #1654, #1649 )

### PR DESCRIPTION
**Pull request by danieldietrich**
Fixes #1651 
Fixes #1662

### CODE REVIEW

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 1]
```diff
@@ -452,14 +456,16 @@ public static void println() {
...
+
     /**
      * Alias for {@link CheckedFunction0#unchecked}
      *
      * @param <R>  return type
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction0}
      */
-    public static <R> Function0<R> Unchecked(CheckedFunction0<R> f) {
+    public static <R> Function0<R> unchecked(CheckedFunction0<R> f) {
```
Comment (danieldietrich) [inline javaslang/src-gen/main/java/javaslang/API.java]: `unchecked` is a helper, not a type. Therefore it starts lower-case

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 2]
```diff
@@ -829,30 +843,7 @@ public static Tuple0 Tuple() {
...
-     *
-     * @param <T>             The value type of a successful result.
-     * @param executorService An executor service.
-     * @param exception       The reason why it failed.
-     * @return A failed {@link Future}.
-     * @throws NullPointerException if executorService or exception is null
-     */
-    public static <T> Future<T> Future(ExecutorService executorService, Throwable exception) {
-        return Future.failed(executorService, exception);
-    }
```
Comment (danieldietrich) [inline javaslang/src-gen/main/java/javaslang/API.java]: this is ambiguous to `Future(T)`

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 3]
```diff
@@ -916,8 +913,9 @@ public static Tuple0 Tuple() {
      * @param value A value.
      * @return A new {@link Try.Success}.
      */
-    public static <T> Try<T> Success(T value) {
-        return Try.success(value);
+    @SuppressWarnings("unchecked")
+    public static <T> Try.Success<T> Success(T value) {
+        return (Try.Success<T>) Try.success(value);
```
Comment (danieldietrich) [inline javaslang/src-gen/main/java/javaslang/API.java]: We need the concrete type towards pattern matching 3.0.0:

```java
List(Try(() -> 1/0)).collect( Case(Success(1), "one") )
```

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 4]
```diff
@@ -1023,37 +1030,15 @@ public static CharSeq CharSeq(CharSequence sequence) {
...
-     * Alias for {@link Array#ofAll(java.util.stream.Stream)}
-     *
-     * @param <T>      Component type of elements.
-     * @param elements Zero or more elements.
-     * @return A new {@link Array} instance containing the given elements
-     * @throws NullPointerException if {@code elements} is null
-     */
-    public static <T> Array<T> Array(java.util.stream.Stream<? extends T> elements) {
-        return Array.ofAll(elements);
-    }
```
Comment (danieldietrich) [inline javaslang/src-gen/main/java/javaslang/API.java]: ambiguous API

File: javaslang/src/main/java/javaslang/collection/PriorityQueue.java [hunk 5]
```diff
@@ -128,7 +128,7 @@ public boolean isOrdered() {
      * @param <T> Component type
      * @return The empty PriorityQueue.
      */
-    public static <T extends Comparable<T>> PriorityQueue<T> empty() {
+    public static <T extends Comparable<? super T>> PriorityQueue<T> empty() {
```
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/collection/PriorityQueue.java]: see http://stackoverflow.com/a/18163768/1110815

File: javaslang/src/test/java/javaslang/collection/VectorPropertyTest.java [hunk 6]
```diff
@@ -229,7 +229,7 @@ public void shouldBehaveLikeArray() {
                 Seq<Tuple2<Seq<Object>, Vector<Object>>> history = Array.empty();
 
                 if (percent(random) < 20) {
-                    expected = Array.ofAll(Vector(randomValues(random, 100)).filter(v -> v instanceof Integer));
+                    expected = Array.ofAll(Vector(randomValues(random, 100)).filter(v -> ((Object) v) instanceof Integer));
```
Comment (danieldietrich) [inline javaslang/src/test/java/javaslang/collection/VectorPropertyTest.java]: did not compile (any more!?) without that cast

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 7]
```diff
@@ -988,72 +992,128 @@ public static CharSeq CharSeq(CharSequence sequence) {
...
-     * @return A new {@link Array} instance containing the given elements
+     * @return A new {@link List} instance containing the given elements
      * @throws NullPointerException if {@code elements} is null
      */
-    public static <T> Array<T> Array(java.util.stream.Stream<? extends T> elements) {
-        return Array.ofAll(elements);
+    @SuppressWarnings("varargs")
+    @SafeVarargs
+    public static <T> Seq<T> Seq(T... elements) {
+        return List.of(elements);
```
Comment (danieldietrich) [inline javaslang/src-gen/main/java/javaslang/API.java]: Scala:

```
scala> Seq()
res0: Seq[Nothing] = List()
```

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 8]
```diff
@@ -1586,240 +1465,69 @@ public static CharSeq CharSeq(CharSequence sequence) {
...
      * @param v1  The value of the 1st pair
      * @param k2  The key of the 2nd pair
      * @param v2  The value of the 2nd pair
-     * @return A new {@link LinkedHashMap} instance containing the given entries
+     * @return A new {@link HashMap} instance containing the given entries
      */
-    public static <K, V> LinkedHashMap<K, V> LinkedMap(K k1, V v1, K k2, V v2) {
-        return LinkedHashMap.of(k1, v1, k2, v2);
+    public static <K, V> Map<K, V> Map(K k1, V v1, K k2, V v2) {
+        return HashMap.of(k1, v1, k2, v2);
```
Comment (danieldietrich) [inline javaslang/src-gen/main/java/javaslang/API.java]: Scala:

```
scala> Map(1 -> "a", 2 -> "b")
res0: scala.collection.immutable.Map[Int,String] = Map(1 -> a, 2 -> b)
```

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 9]
```diff
@@ -829,30 +842,7 @@ public static Tuple0 Tuple() {
         return Future.successful(executorService, result);
     }
```
Comment (danieldietrich) [inline javaslang/src-gen/main/java/javaslang/API.java]: Scala:

```
scala> Future(1)
res0: scala.concurrent.Future[Int] = scala.concurrent.impl.Promise$DefaultPromise@57404a8d

scala> Future(1/0)
res1: scala.concurrent.Future[Int] = scala.concurrent.impl.Promise$DefaultPromise@6cddb04c
```

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 10]
```diff
@@ -471,7 +476,7 @@ public static void println() {
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction1}
      */
-    public static <T1, R> Function1<T1, R> Unchecked(CheckedFunction1<T1, R> f) {
+    public static <T1, R> Function1<T1, R> unchecked(CheckedFunction1<T1, R> f) {
```
Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T1' must match pattern '^\[A\-Z\]$'\.

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 11]
```diff
@@ -484,7 +489,7 @@ public static void println() {
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction2}
      */
-    public static <T1, T2, R> Function2<T1, T2, R> Unchecked(CheckedFunction2<T1, T2, R> f) {
+    public static <T1, T2, R> Function2<T1, T2, R> unchecked(CheckedFunction2<T1, T2, R> f) {
```
Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T1' must match pattern '^\[A\-Z\]$'\.

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 12]
```diff
@@ -498,7 +503,7 @@ public static void println() {
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction3}
      */
-    public static <T1, T2, T3, R> Function3<T1, T2, T3, R> Unchecked(CheckedFunction3<T1, T2, T3, R> f) {
+    public static <T1, T2, T3, R> Function3<T1, T2, T3, R> unchecked(CheckedFunction3<T1, T2, T3, R> f) {
```
Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T1' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T2' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T3' must match pattern '^\[A\-Z\]$'\.

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 13]
```diff
@@ -513,7 +518,7 @@ public static void println() {
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction4}
      */
-    public static <T1, T2, T3, T4, R> Function4<T1, T2, T3, T4, R> Unchecked(CheckedFunction4<T1, T2, T3, T4, R> f) {
+    public static <T1, T2, T3, T4, R> Function4<T1, T2, T3, T4, R> unchecked(CheckedFunction4<T1, T2, T3, T4, R> f) {
```
Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T1' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T2' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T3' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T4' must match pattern '^\[A\-Z\]$'\.

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 14]
```diff
@@ -529,7 +534,7 @@ public static void println() {
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction5}
      */
-    public static <T1, T2, T3, T4, T5, R> Function5<T1, T2, T3, T4, T5, R> Unchecked(CheckedFunction5<T1, T2, T3, T4, T5, R> f) {
+    public static <T1, T2, T3, T4, T5, R> Function5<T1, T2, T3, T4, T5, R> unchecked(CheckedFunction5<T1, T2, T3, T4, T5, R> f) {
```
Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T1' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T2' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T3' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T4' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T5' must match pattern '^\[A\-Z\]$'\.

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 15]
```diff
@@ -546,7 +551,7 @@ public static void println() {
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction6}
      */
-    public static <T1, T2, T3, T4, T5, T6, R> Function6<T1, T2, T3, T4, T5, T6, R> Unchecked(CheckedFunction6<T1, T2, T3, T4, T5, T6, R> f) {
+    public static <T1, T2, T3, T4, T5, T6, R> Function6<T1, T2, T3, T4, T5, T6, R> unchecked(CheckedFunction6<T1, T2, T3, T4, T5, T6, R> f) {
```
Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T1' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T2' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T3' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T4' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T5' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T6' must match pattern '^\[A\-Z\]$'\.

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 16]
```diff
@@ -564,7 +569,7 @@ public static void println() {
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction7}
      */
-    public static <T1, T2, T3, T4, T5, T6, T7, R> Function7<T1, T2, T3, T4, T5, T6, T7, R> Unchecked(CheckedFunction7<T1, T2, T3, T4, T5, T6, T7, R> f) {
+    public static <T1, T2, T3, T4, T5, T6, T7, R> Function7<T1, T2, T3, T4, T5, T6, T7, R> unchecked(CheckedFunction7<T1, T2, T3, T4, T5, T6, T7, R> f) {
```
Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T5' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T7' must match pattern '^\[A\-Z\]$'\.

File: javaslang/src-gen/main/java/javaslang/API.java [hunk 17]
```diff
@@ -583,10 +588,12 @@ public static void println() {
      * @param f    A method reference
      * @return A unchecked wrapper of supplied {@link CheckedFunction8}
      */
-    public static <T1, T2, T3, T4, T5, T6, T7, T8, R> Function8<T1, T2, T3, T4, T5, T6, T7, T8, R> Unchecked(CheckedFunction8<T1, T2, T3, T4, T5, T6, T7, T8, R> f) {
+    public static <T1, T2, T3, T4, T5, T6, T7, T8, R> Function8<T1, T2, T3, T4, T5, T6, T7, T8, R> unchecked(CheckedFunction8<T1, T2, T3, T4, T5, T6, T7, T8, R> f) {
```
Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T2' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T3' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T4' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T5' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T7' must match pattern '^\[A\-Z\]$'\.

Comment (sputnikci) [inline javaslang/src-gen/main/java/javaslang/API.java]: \[Checkstyle\] ERROR: Name 'T8' must match pattern '^\[A\-Z\]$'\.

**Comment by danieldietrich** [comment]
I must repair that CheckStyle.config o_O

**Pull request merged**